### ISSUE TITLE
non-blocking rate-limiting

### ISSUE THREAD

**Issue by ma1ko** [brayniac/ratelimit#30]
Currently the only way get a tocken is calling ```wait()```. I need a non-blocking version (maybe ```get()```) that tries to get a token, but if none is available the method returns with an error instead of waiting
The use-case here is aborting operations if they are over their limit instead of waiting for new tokens.

Is this possible to implement?

**Comment by brayniac** [comment]
Yes. This is already implemented. ```.try_wait()``` and ```.try_wait_for()```. Looks like the doc coverage isn't there for these. Will leave this open to address that.

**Comment by brayniac** [comment]
Awesome! I'll fix-up the documentation. Thanks for opening this!
